src/classes/OPP_CreateTribute_BATCH.cls

```diff
@@ -0,0 +1,129 @@
+    }
+
+    /******************************************************************************************************
+    * @description Batch start method. Runs through all the Opportunities to be converted.
+    */
+    public Database.QueryLocator start(Database.BatchableContext info) {
+        String soqlQuery = ((additionalWhereClause != null) && (additionalWhereClause.trim().length() > 0)) ?
+                            OPP_QUERY + ' WHERE (' + additionalWhereClause + ')' : OPP_QUERY;
+
+        // Deliberately not trapping this potentially invalid query (since Where Clause might be invalid).
+        // This will display an error in the job widget used on the settings page, and send the person who 
+        // launched it a script error.  Decided this was best option with Nicolas Campbell
+        return Database.getQueryLocator(soqlQuery);
+    }
+
+    /*******************************************************************************************************
+    * @description Batch execute method.
+    * @return void.
+    ********************************************************************************************************/
+    public void execute(Database.BatchableContext BC, List<Opportunity> scope) {
+
+        List<Tribute__c> newTributes = new List<Tribute__c>();
+        for (Opportunity opp : scope) {
+            if (hasTributeInfo(opp)) {
+                newTributes.add(
+                    new Tribute__c(
+                        Opportunity__c = opp.Id,
+                        Tribute_Type__c = opp.Tribute_Type__c,
+                        Honoree_Contact__c = opp.Honoree_Contact__c,
+                        Honoree_Name__c = opp.Honoree_Name__c
+                    )
+                );
+
+                //clear the Opportunity fields
+                opp.Tribute_Type__c = null;
+                opp.Honoree_Contact__c = null;
+                opp.Honoree_Name__c = null;
+            }
+        }
+        
+        //We do not want to run through Opportunity or Tribute triggers when clearing Tribute Fields
+        //on the Opportunity
+        TDTM_TriggerHandler.disableTDTM = true;
+        insert newTributes;
+        update scope;
+        TDTM_TriggerHandler.disableTDTM = false;
+    }
+
+    /*******************************************************************************************************
+    * @description Checks to see if there is Tribute Information on the Opportunity
+    * It returns true if one or more of the following fields null/blank: Honoree_Contact__c (null only),
+    * Honoree_Name__c (null/blank), Tribute_Type__c (null/blank).
+    * @param Opportunity opp - The opportunity to check
+    * @return Boolean true if Tribute info is present, false otherwise
+    ********************************************************************************************************/
+    @testVisible private static Boolean hasTributeInfo(Opportunity opp) {
+        return ((opp.Honoree_Contact__c != null) || 
```

Comment:
I think this might be easier to read using isNotBlank(): `String.IsNotBlank(opp.Honoree_Contact__c)`
https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_methods_system_string.htm